File: sg_cdez_web/src/services/perfilService.ts
import { apiClient } from "../utils/apiHelper";

import type {
  PerfilActualizarRequest,
  PerfilResponse,
} from "./interfaces/perfilInterface";


export async function obtenerPerfil(): Promise<PerfilResponse> {
  const response = await apiClient.get<PerfilResponse>(
    "/perfil",
  );

  return response.data;
}


export async function actualizarPerfil(
  request: PerfilActualizarRequest,
): Promise<PerfilResponse> {
  const response = await apiClient.put<PerfilResponse>(
    "/perfil",
    request,
  );

  return response.data;
}

File: sg_cdez_api/src/main/java/com/cdez/sg_cdez_api/service/PerfilService.java
package com.cdez.sg_cdez_api.service;

import com.cdez.sg_cdez_api.dto.request.PerfilActualizarRequest;
import com.cdez.sg_cdez_api.dto.response.PerfilResponse;

public interface PerfilService {

    PerfilResponse obtenerPerfil();

    PerfilResponse actualizarPerfil(PerfilActualizarRequest request);
}